packages/graphql-transformer-core/src/Transformer.ts

```diff
@@ -15,36 +16,17 @@ import {
     EnumValueDefinitionNode,
     TypeDefinitionNode,
     DefinitionNode,
-    ArgumentNode,
-    valueFromASTUntyped,
     DocumentNode
 } from 'graphql'
 import { InvalidTransformerError } from './errors'
 
-function reduceTypeDefinitionNodes(
-    acc: { [name: string]: TypeDefinitionNode },
-    definition: DefinitionNode
-): { [name: string]: TypeDefinitionNode } {
-    switch (definition.kind) {
-        case 'ScalarTypeDefinition':
-        case 'ObjectTypeDefinition':
-        case 'InterfaceTypeDefinition':
-        case 'UnionTypeDefinition':
-        case 'EnumTypeDefinition':
-        case 'InputObjectTypeDefinition':
-            return { ...acc, [definition.name.value]: definition }
-        default:
-            return acc;
-    }
-}
-
 /**
  * A GraphQLTransformer takes a context object, processes it, and
  * returns a new context. The transformer is responsible for returning
  * a context that fully describes the infrastructure requirements
  * for its stage of the transformation.
  */
-export default class Transformer {
+export default class Transformer implements ITransformer {
```

Comment:
When writing transformers you may extend this class but now it is optional as you can use plain JS objects as well.